filesystem/replicator_test.go

```diff
@@ -136,4 +136,33 @@ import (
 		}
 	 })
  }
+
+ func TestReplicateAdditionCallBack(t *testing.T) {
+	t.Run("test for destination dir is removed", func(t *testing.T) {
+		sourceDir := "testdata/source"
+		destinationDir := "testdata/destination"
+		destinationFilePath := destinationDir + "/existing-file.txt"
+		// Prepare the destination directory with an existing file
+		err := os.MkdirAll(destinationDir, 0755)
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, err = os.Create(destinationFilePath)
+		 if err != nil {
+			 t.Fatal(err)
+		 }
+
+		// Call the addition callback
+		err = replicateAdditionCallBack(sourceDir, destinationDir, nil)
+		if err != nil {
+			t.Fatalf("Expected no error, got %s", err)
+		}
+
+		// Check if the destination directory is removed
+		_, err = os.Stat(destinationDir)
```

Comment:
Could you please check for specific to directory deletion? You can also retain the generic error check too.